node/ipfs.go

```diff
@@ -0,0 +1,174 @@
+package node
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"path/filepath"
+	"sync"
+
+	ipfscfg "github.com/ipfs/go-ipfs-config"
+	oldcmds "github.com/ipfs/go-ipfs/commands"
+	ipfscore "github.com/ipfs/go-ipfs/core"
+	"github.com/ipfs/go-ipfs/core/coreapi"
+	"github.com/ipfs/go-ipfs/core/corehttp"
+	"github.com/ipfs/go-ipfs/core/node/libp2p"
+	"github.com/ipfs/go-ipfs/plugin/loader"
+	"github.com/ipfs/go-ipfs/repo"
+	"github.com/ipfs/go-ipfs/repo/fsrepo"
+	coreiface "github.com/ipfs/interface-go-ipfs-core"
+	"github.com/jbenet/goprocess"
+	ma "github.com/multiformats/go-multiaddr"
+	manet "github.com/multiformats/go-multiaddr/net"
+
+	cfg "github.com/lazyledger/lazyledger-core/config"
+	"github.com/lazyledger/lazyledger-core/libs/log"
+	"github.com/lazyledger/lazyledger-core/p2p/ipld/plugin"
+)
+
+// IPFSAPIProvider allows customizable IPFS core APIs.
+type IPFSAPIProvider func(*cfg.Config, log.Logger) (coreiface.CoreAPI, io.Closer, error)
+
+// DefaultIPFSAPIProvider is the provider that embeds IPFS node within the same process.
+// It also returns closable for graceful node shutdown.
+func DefaultIPFSAPIProvider(cfg *cfg.Config, logger log.Logger) (coreiface.CoreAPI, io.Closer, error) {
```

Comment:
Now that the API only contains the repoRoot (and a boolean flag ServeHttpAPI), should we just pass those in? That makes the code more self-contained. For the light-client, we might want to move this to another package than node. Not depending on `cfg` will make that easier.

Reply:
Check out new approach